src/groupeddataframe/grouping.jl

```diff
@@ -412,28 +412,53 @@ true
 """
 Base.keys(gd::GroupedDataFrame) = GroupKeys(gd)
 
-# Index with GroupKey
-function Base.getindex(gd::GroupedDataFrame, key::GroupKey)
-    gd === parent(key) && return gd[getfield(key, :idx)]
+
+# The allowed key types for dictionary-like indexing
+const AbstractGroupKey = Union{GroupKey, Tuple, NamedTuple}
```

Comment:
maybe better call it `GroupKeyUnion`, as "abstract" suggests abstract type.

@nalimilan - any opinion?

Reply:
That makes sense, it's only used internally anyway.